## **Logistic Regression Code Implementation**

First of all, we will need to import the necessary libraries required for implementing the algorithm. 


In [1]:
#importing pandas which is used for data analysis and manipulation
import pandas as pd
#importing the logistic regression module from sklearn.linear_model
from sklearn.linear_model import LogisticRegression
# for splitting od X and y into training and testing sets
from sklearn.model_selection import train_test_split
#importing confusion matrix which is used for evaluation of the performance of a classification model
from sklearn.metrics import confusion_matrix


Now we import the dataset.



In [3]:
#importing the dataset
dataset = pd.read_csv("diabetes.csv")

In [4]:
dataset.head()

Pregnancies  Glucose  BloodPressure  ...  DiabetesPedigreeFunction  Age  Outcome
0            6      148             72  ...                     0.627   50        1
1            1       85             66  ...                     0.351   31        0
2            8      183             64  ...                     0.672   32        1
3            1       89             66  ...                     0.167   21        0
4            0      137             40  ...                     2.288   33        1

[5 rows x 9 columns]

No we define the input features which are used for the predicting the classification label given by 'Outcome' here.

In [5]:
input_features=['Pregnancies', 'Insulin', 'BMI', 'Age','Glucose','BloodPressure','DiabetesPedigreeFunction','SkinThickness']
X=dataset[input_features]
Y=dataset['Outcome']

Now we do the splitting of the data into training data and testing data with 80% as training and 20% as testing data.

In [6]:
X_train,X_test,y_train,y_test=train_test_split(X,Y,test_size=0.20,random_state=0)

Then we define the model for logistic regression with the accurate parameters.

In [7]:
model = LogisticRegression(solver='lbfgs',class_weight='balanced', max_iter=10000)

Fitting the model

In [8]:
model.fit(X_train,y_train)

LogisticRegression(C=1.0, class_weight='balanced', dual=False,
                   fit_intercept=True, intercept_scaling=1, l1_ratio=None,
                   max_iter=10000, multi_class='auto', n_jobs=None,
                   penalty='l2', random_state=None, solver='lbfgs', tol=0.0001,
                   verbose=0, warm_start=False)

Now we get the predicted values using logistic regression model in y_pred.

In [9]:
y_pred=model.predict(X_test)

A confusion matrix is a table that is used to evaluate the performance of a classification model. So we include the same in our code.

In [10]:
# import the metrics class
from sklearn import metrics
confusion_matrix = metrics.confusion_matrix(y_test, y_pred)
print(confusion_matrix)

[[82 25]
 [11 36]]


Accuracy is a metric which gives the ratio of correct predictions out of the total no. of predictions.

In [15]:
score = model.score(X_test, y_test)
print(score)

0.7662337662337663


So, we get an accuracy of 76.62 % on applying logistic regression.